File: 110-balanced-binary-tree/balanced-binary-tree.js
/**
 * Definition for a binary tree node.
 * function TreeNode(val, left, right) {
 *     this.val = (val===undefined ? 0 : val)
 *     this.left = (left===undefined ? null : left)
 *     this.right = (right===undefined ? null : right)
 * }
 */
/**
 * @param {TreeNode} root
 * @return {boolean}
 */
var isBalanced = function (root) {

    let ans = true;
    const rec = (curr) => {
        if (!curr) return 0;
        let left = rec(curr.left);
        let right = rec(curr.right);
        if (Math.abs(left - right) > 1) ans = false;
        return 1 + Math.max(left, right)
    }
    rec(root);
    return ans;
};

File: 110-balanced-binary-tree/balanced-binary-tree.java
/**
 * Definition for a binary tree node.
 * public class TreeNode {
 *     int val;
 *     TreeNode left;
 *     TreeNode right;
 *     TreeNode() {}
 *     TreeNode(int val) { this.val = val; }
 *     TreeNode(int val, TreeNode left, TreeNode right) {
 *         this.val = val;
 *         this.left = left;
 *         this.right = right;
 *     }
 * }
 */
class Solution {
    boolean ans = true;
    public boolean isBalanced(TreeNode root) {
        rec(root);
        return ans;
    }
    public int rec( TreeNode curr) {
        if( curr == null ) return 0;
        int left = rec(curr.left);
        int right = rec(curr.right);

        if( Math.abs(left - right) > 1) {
            ans = false;
        }
        return 1 + Math.max(left, right);
    }
}